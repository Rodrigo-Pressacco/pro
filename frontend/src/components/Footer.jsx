function Footer(){
    return(
        <footer className="footer">
            <p>Â©2021 Neo Travel</p>
            <ul className="social-network">
                <li>
                    <img src="../images/facebook.png" alt="facebook" />
                </li>
                <li>
                    <img src="../images/linkedin.png" alt="linkedin" />
                </li>
                <li>
                    <img src="../images/twitter.png" alt="twitter" />
                </li>
                <li>
                    <img src="../images/instagram.png" alt="instagram" />
                </li>
            </ul>
        </footer>
    )
}
export default Footer;