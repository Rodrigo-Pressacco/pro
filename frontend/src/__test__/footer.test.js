import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Footer from '../components/Footer'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));


describe('<Footer/>', () => {
    let footer; 

    beforeEach(() => {
        footer = shallow(<Footer />)
    });

    it("Deberia mostrar <Footer/> correctamente", () => {
        expect(footer).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta p", () => {
        let node = footer.find("p")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta ul", () => {
        let node = footer.find("ul")
        expect(node.length).toEqual(1)
    });
    
    it("Deberia retornar 4 etiquetas li", () => {
        let node = footer.find("li")
        expect(node.length).toEqual(4)
    });

    it("Deberia retornar 4 etiquetas img", () => {
        let node = footer.find("img")
        expect(node.length).toEqual(4)
    });

});



