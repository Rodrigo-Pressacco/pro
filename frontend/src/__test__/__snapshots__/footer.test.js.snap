// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> Deberia mostrar <Footer/> correctamente 1`] = `
<footer
  className="footer"
>
  <p>
    Â©2021 Neo Travel
  </p>
  <ul
    className="social-network"
  >
    <li>
      <img
        alt="facebook"
        src="../images/facebook.png"
      />
    </li>
    <li>
      <img
        alt="linkedin"
        src="../images/linkedin.png"
      />
    </li>
    <li>
      <img
        alt="twitter"
        src="../images/twitter.png"
      />
    </li>
    <li>
      <img
        alt="instagram"
        src="../images/instagram.png"
      />
    </li>
  </ul>
</footer>
`;
